packages/subcommands/src/lib/types/Events.ts

```diff
@@ -0,0 +1,94 @@
+	MessageSubcommandDenied: 'messageSubcommandDenied' as const,
+	MessageSubcommandError: 'messageSubcommandError' as const
+};
+
+export interface IMessageSubcommandPayload {
+	message: Message;
+	command: MessageCommand;
+}
+
+export interface MessageSubcommandAcceptedPayload extends IMessageSubcommandPayload {
+	context: MessageCommand.RunContext;
+}
+
+export interface MessageSubcommandRunPayload extends MessageSubcommandAcceptedPayload {}
+
+export interface MessageSubcommandErrorPayload extends MessageSubcommandRunPayload {}
+
+export interface MessageSubcommandDeniedPayload extends MessageSubcommandRunPayload {
+	parameters: string;
+	subcommand: MessageSubcommandMappingValue;
+}
+
+export interface MessageSubcommandSuccessPayload extends MessageSubcommandRunPayload {
+	result: unknown;
+}
+
+export interface IChatInputSubcommandPayload {
+	interaction: ChatInputCommand.Interaction;
+	command: ChatInputCommand;
+}
+
+export interface ChatInputSubcommandAcceptedPayload extends IChatInputSubcommandPayload {
+	context: ChatInputCommand.RunContext;
+}
+
+export interface ChatInputSubcommandRunPayload extends ChatInputSubcommandAcceptedPayload {}
+
+export interface ChatInputSubcommandErrorPayload extends ChatInputSubcommandRunPayload {}
+
+export interface ChatInputSubcommandDeniedPayload extends ChatInputSubcommandRunPayload {
+	subcommand: ChatInputSubcommandMappingValue;
+}
+
+export interface ChatInputSubcommandSuccessPayload extends ChatInputSubcommandRunPayload {
+	result: unknown;
+}
+
+declare module 'discord.js' {
+	interface ClientEvents {
+		[Events.ChatInputSubcommandRun]: [
+			interaction: Interaction,
+			subcommand: ChatInputSubcommandMappingValue,
+			payload: ChatInputSubcommandRunPayload
+		];
+		[Events.ChatInputSubcommandSuccess]: [
+			interaction: Interaction,
+			subcommand: ChatInputSubcommandMappingValue,
+			payload: ChatInputSubcommandSuccessPayload
+		];
+		[Events.ChatInputSubcommandNotFound]: [
+			interaction: Interaction,
+			subcommand: ChatInputSubcommandMappingValue,
+			context: ChatInputCommand.Context
+		];
+		[Events.ChatInputSubcommandDenied]: [error: UserError, payload: ChatInputSubcommandDeniedPayload];
+		[Events.ChatInputSubcommandError]: [error: unknown, payload: ChatInputSubcommandErrorPayload];
+		[Events.MessageSubcommandRun]: [message: Message, subcommand: MessageSubcommandMappingValue, payload: MessageSubcommandRunPayload];
+		[Events.MessageSubcommandSuccess]: [message: Message, subcommand: MessageSubcommandMappingValue, payload: MessageSubcommandSuccessPayload];
+		[Events.MessageSubcommandNotFound]: [message: Message, subcommand: MessageSubcommandMappingValue, context: ChatInputCommand.Context];
+		[Events.MessageSubcommandDenied]: [error: UserError, payload: MessageSubcommandDeniedPayload];
+		[Events.MessageSubcommandError]: [error: unknown, payload: MessageSubcommandErrorPayload];
+	}
+}
+
+declare module '@sapphire/framework' {
+	const enum Identifiers {
```

Comment:
This won't be present at runtime for JS users, so please add it to the enum too 😓